presto-main/src/main/java/com/facebook/presto/operator/scalar/Re2JRegexpFunctions.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.operator.Description;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.type.Re2JRegexp;
+import com.facebook.presto.type.Re2JRegexpType;
+import com.facebook.presto.type.SqlType;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import javax.annotation.Nullable;
+
+public final class Re2JRegexpFunctions
+{
+    private Re2JRegexpFunctions()
+    {
+    }
+
+    @Description("returns substrings matching a regular expression")
+    @ScalarFunction
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean regexpLike(@SqlType(StandardTypes.VARCHAR) Slice source, @SqlType(Re2JRegexpType.NAME) Re2JRegexp pattern)
+    {
+        return pattern.matches(source);
+    }
+
+    @Description("removes substrings matching a regular expression")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice regexpReplace(@SqlType(StandardTypes.VARCHAR) Slice source, @SqlType(Re2JRegexpType.NAME) Re2JRegexp regexpGenericPattern)
```

Comment:
rename `regexpGenericPattern`s in this class. This is no longer an appropriate name.